pulsar-client/src/main/java/org/apache/pulsar/client/impl/PatternMultiTopicsConsumerImpl.java

```diff
@@ -69,6 +69,31 @@ public class PatternMultiTopicsConsumerImpl<T> extends MultiTopicsConsumerImpl<T
     private volatile Timeout recheckPatternTimeout = null;
     private volatile String topicsHash;
 
+    /**
+     * This variable is used to make all tasks that will modify subscriptions will be executed one by one.
```

Comment:
Interesting solution. Does the current solution of chaining the futures introduce a memory leak?

One possible solution for limiting concurrency of asynchronous calls is to use [ConcurrencyReducer from com.spotify:completable-futures](https://github.com/spotify/completable-futures/blob/master/src/main/java/com/spotify/futures/ConcurrencyReducer.java).  

In this case, limiting concurrency would be useful for a few locations where all asynchronous operations happen concurrently. For example, the code where `consumer.closeAsync()` is called happens all at once. 
It might not be a problem in this case since binary protocol calls will go on the same set of connections. For Pulsar Admin API calls, there are problems for example in topic deletion where concurrency isn't limited (problem explained in https://github.com/apache/pulsar/pull/22541#issuecomment-2071568113).

Reply:
Answering the memory leak question myself: it seems that memory leaks with chaining have been addressed a long time ago in the JVM: https://stackoverflow.com/a/73157976 . It should be fine to have the solution.

Reply:
Added a tool named `PatternConsumerUpdateQueue` to fix this issue